### PR TITLE
feat: introduce post processor for backlink audit SITES-18417

### PR DESCRIPTION
**Pull request by dzehnder**
# PR Status
This PR implements the post processor for [SITES-18417](https://jira.corp.adobe.com/browse/SITES-18417).
Relates to https://github.com/adobe/spacecat-audit-worker/pull/77

### CODE REVIEW

File: src/backlinks/handler.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+ * governing permissions and limitations under the License.
+ */
+
+import { badRequest, internalServerError, noContent } from '@adobe/spacecat-shared-http-utils';
+import { hasText, isArray, isObject } from '@adobe/spacecat-shared-utils';
+import {
+  markdown, postSlackMessage, section, uploadSlackFile,
+} from '../support/slack.js';
+
+function convertToCSV(array) {
```
Comment (solaris007) [inline src/backlinks/handler.js:19]: could be made more efficient and robust like this:
```suggestion
function convertToCSV(array) {
  if (array.length === 0) {
    return '';
  }

  const headers = Object.keys(array[0]).join(',');
  const rows = array.map(item => {
    return Object.values(item).map(value => {
      if (typeof value === 'object' && value !== null) {
        return `"${JSON.stringify(value)}"`;
      }
      return `"${value}"`;
    }).join(',');
  }).join('\r\n');

  return `${headers}\r\n${rows}\r\n`;
}
```

File: src/backlinks/handler.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+      return `"${value}"`;
+    }).join(',');
+
+    csvStr += `${line}\r\n`;
+  });
+
+  return csvStr;
+}
+
+function buildSlackMessage(url, fileUrl, fileName, auditResult) {
```
Comment (solaris007) [inline src/backlinks/handler.js:43]: could be simplified as:
```suggestion
function buildSlackMessage(url, fileUrl, fileName, auditResult) {
  return [
    section({
      text: markdown(`For *${url}*, ${auditResult.brokenBacklinks.length} broken backlink(s) were detected.`),
    }),
    section({
      text: markdown(`The following CSV file contains a detailed report for all broken backlinks: <${fileUrl}|${fileName}>.`),
    }),
  ];
}
```

File: src/backlinks/handler.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  blocks.push((section({
+    text: markdown(`The following CSV file contains a detailed report for all broken backlinks: <${fileUrl}|${fileName}>.`),
+  })));
+
+  return blocks;
+}
+function isValidMessage(message) {
+  return hasText(message.url)
+    && isArray(message.auditResult?.brokenBacklinks)
+    && isObject(message.auditContext?.slackContext);
```
Comment (solaris007) [inline src/backlinks/handler.js:59]: should we add here?
```
    && message.auditResult.brokenBacklinks.every(link => isObject(link))

```

**Review by solaris007 — APPROVED** [review]

File: src/backlinks/handler.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    }
+    return `"${value}"`;
+  }).join(',')).join('\r\n');
+  return `${headers}\r\n${rows}\r\n`;
+}
+
+function buildSlackMessage(url, fileUrl, fileName, auditResult) {
+  return [
+    section({
+      text: markdown(`For *${url}*, ${auditResult.brokenBacklinks.length} broken backlink(s) were detected.`),
```
Comment (ekremney) [inline src/backlinks/handler.js:33]: for the sake of readability for customers and consistency with other alerts/reports we have, I suggest to strip the `https://` part from the url label here. For example:


For [adobe.com](http://adobe.com/), 17 page(s) had 404s last week for the real users.

https://cq-dev.slack.com/archives/C060T2PPF8V/p1705636837033839?thread_ts=1705636823.626089&cid=C060T2PPF8V

**Review by ekremney — APPROVED** [review]

**Pull request merged**